modules/apps/client-extension/client-extension-web/test/js/AttribureFields.test.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {fireEvent, render} from '@testing-library/react';
+import React from 'react';
+
+import AttributeFields from '../../src/main/resources/META-INF/resources/js/components/global-js/AttributeFields.tsx';
+
+describe('AttributeFields', () => {
+	it('Prevents user from adding space on attributes', () => {
+		const onAttributeChange = jest.fn();
+
+		const {getByTestId} = render(
+			<AttributeFields
+				index={0}
+				onAttributeChange={onAttributeChange}
+				value=""
+			/>
+		);
+
+		const attributeField = getByTestId('name');
+
+		fireEvent.change(attributeField, {target: {value: 'name with space'}});
```

Comment:
It looks like this test is using normal space characters. We need to make sure that we are testing the actual bug, which requires us to test `&nbsp;`. So it would be good to test both a normal space and a non-breaking-space.